🐛 BUG: self-closing <img /> tag in <Markdown> in astro@next--compiler causes panic

### What package manager are you using?

pnpm

### What operating system are you using?

Linux

### Describe the Bug

Using astro `0.0.0-compiler-2021928195510`

Use the following `.astro` page:

```
---
import { Markdown } from 'astro/components';

const img = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAMAAAADCAIAAADZSiLoAAAAJElEQVR42gXBQQEAIAgEsGn/JCTwRx3eVwA3G8qGnmz05OI8+OMgDGtMzxoHAAAAAElFTkSuQmCC";
---
<html>
  <body>
    <Markdown>
      <img src={img} />

      ```
      a brace is incoming {
      ```
    </Markdown>
  </body>
</html>
```

The `astro dev` process gives

```
panic: runtime error: index out of range
```

and the output page says

```
500 Internal Error
unreachable
```

Notes:

Each of these variations individually make it work:

* Replacing `<img src={img} />` with `<img src={img}></img>`.
* Removing the `{` in `a brace is incoming {`
* Removing all the backticks
  * However in this case, the `{` is rendered as a `}`

Also the example works without any modifications in astro 0.20.6, the output being

```
(very small image)
a brace is incoming {
```

### Steps to Reproduce

1. `npm init astro` using any template
2. `npm install astro@next--compiler`
3. Use the astro file above
4. Error as above, would like to see behaviour from 0.20.6 maintained

### Link to Minimal Reproducible Example (Optional)

_No response_

--- a/internal/token_test.go
+++ b/internal/token_test.go
@@ -197,6 +197,15 @@ func TestBasic(t *testing.T) {
 			`<Fragment>foo</Fragment>`,
 			[]TokenType{StartTagToken, TextToken, EndTagToken},
 		},
+		{
+			"Markdown codeblock",
+			fmt.Sprintf(`<Markdown>
+%s%s%s
+open brace {
+%s%s%s
+</Markdown>`, "`", "`", "`", "`", "`", "`"),
+			[]TokenType{StartTagToken, TextToken, TextToken, TextToken, TextToken, EndTagToken},
+		},
 	}
 
 	runTokenTypeTest(t, Basic)

